### DISCUSSION TITLE
How to add multiple values to csp directives

### DISCUSSION THREAD

**Discussion by discoverlance-com** [KindSpells/astro-shield#77] [Q&A] [answered]
First of all, thanks for this package. Really helped me a lot. The example in the website does not specify one where you can use multiple values for one directive so not sure what's the right thing to do as the only example is a single one with 'none'.

I am trying to use it to configure my csp, but I am having issues with the generated csp policy.
This is my csp config:
```js
securityHeaders: {
        contentSecurityPolicy: {
          cspDirectives: {
                'default-src': "'self'",
	        'base-uri': "'self'",
	        'form-action':  "'self'",						
		'script-src':  "'self' 'strict-dynamic' fonts.googleapis.com fonts.gstatic.com https://*.clarity.ms https://c.bing.com", 
		'style-src': "'self' font.googleapis.com",
		'font-src': "'self' fonts.gstatic.com fonts.googleapis.com",
		'img-src': "'self' data:",
		'object-src':  "'none'",
		'frame-ancestors': "'self'"
          }
        }
      }

```

But it seems not to work as I am getting an error in my console.
![image](https://github.com/kindspells/astro-shield/assets/52635726/685d6e6a-c81e-45c8-8991-1c4167df0f0d)

Also, I see that the function that serializes the headers checks for single quotes and if not available, it adds them. I think not all directives need that as ones like `'self' 'strict-dynamic'` request quotes but urls and other values don't. Could that be the issue or I am just getting something wrong here. Please help. Thanks.
```js
// I mean this serialise function.
export const safeSerialiseHashes = hashes =>
	Array.from(hashes)
		.sort()
		.map(h => (h.match(/^'[^']+'$/i) ? h : `'${h}'`))
		.join(' ')
```

When I check the content policy header too, I can see quotes around the urls:

![image](https://github.com/kindspells/astro-shield/assets/52635726/6885a906-21cb-477e-8c77-4a6a8ccf6269)

**Comment by castarco** [comment]
Hi @discoverlance-com , please excuse me for the delay in my response.

I suspect that your problem might be a combination of bad documentation + some subtle bug... I'll have to check it.

**Reply by castarco** [reply, marked as answer]
Hi again @discoverlance-com , I think this release ( https://github.com/kindspells/astro-shield/releases/tag/1.3.6 ) should help.

**Reply by discoverlance-com** [reply]
> Hi again @discoverlance-com , I think this release ( https://github.com/kindspells/astro-shield/releases/tag/1.3.6 ) should help.

Ok will check it out. Thanks.
